src/ngraph/runtime/plaidml/plaidml_ops_batch_norm.cpp

```diff
@@ -0,0 +1,297 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph/log.hpp"
+#include "ngraph/op/batch_norm.hpp"
+#include "ngraph/runtime/plaidml/plaidml_impl.hpp"
+
+namespace vp = vertexai::plaidml;
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace plaidml
+        {
+            // BatchNorm implements batch normalization.
+            template <>
+            void Impl<op::BatchNorm>::operator()()
```

Comment:
A lot of these files would probably look better if we removed the nested namespaces and prepended ngraph::runtime::plaidml:: to the implementations.

Reply:
Done.  (For some reason, I thought you couldn't specialize a template from another namespace; looks like I should've tried it before assuming.  :-)